src/cascadia/TerminalApp/CascadiaSettings.cpp

```diff
@@ -528,6 +528,19 @@ void CascadiaSettings::_ReorderProfilesToMatchUserSettingsOrder()
                     auto guidJson{ profileJson["guid"] };
                     auto guid = Utils::GuidFromString(GetWstringFromJson(guidJson));
 
+                    if (uniqueGuids.insert(guid).second)
+                    {
+                        guidOrder.push_back(guid);
+                    }
+                }
+                else
+                {
+                    // In this case, _we_ generated a GUID for that profile. We
+                    // should similarly generate that guid here, so we can order
+                    // it appropriately.
+                    auto name = GetWstringFromJson(profileJson["name"]);
+                    auto guid = Profile::GenerateGuidForProfile(name);

```

Comment:
This seems fragile, but I understand why we're doing it. What about when you add `source` to the guid generator?

Reply:
Do we actually just need a static method on `Profile` that takes in a json blob and returns a GUID? It'll hide all of the logic inside Profile.

Reply:
I plan on just adding another arg in that function for the source too. It certainly bothers me that we have to have CascadiaSettings doing this much profile-specific deserializing, but I wasn't sure how to do this without this mess.